app/models/merchant.rb

```diff
@@ -51,4 +51,22 @@ def top_5_items
       .order("item_revenue DESC")
       .limit(5)
   end
+
+  def self.top_five_merchants
+    Merchant.joins(items: { invoices: [:invoice_items, :transactions] })
+            .select("merchants.id, merchants.name, SUM(invoice_items.quantity * invoice_items.unit_price) as total_revenue")
+            .where("transactions.result=1")
+            .group("merchants.id")
+            .order("total_revenue DESC")
+            .limit(5)
+  end
+
+  def best_day
```

Comment:
Good use of self in order to get the ID instead of what we did before. Good querying.